gui/dialogs/mcduweb.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<!-- Copyright (c) 2020 Josh Davidson (Octal450) -->
+
+<!-- A3XX MCDU web launcher Dialog -->
+
+<PropertyList>
+
+	<name>mcdu-web</name>
+	<layout>vbox</layout>
+	<pref-width>480</pref-width>
+
+    <group>
+        <layout>hbox</layout>
+        <text>
+            <halign>left</halign>
+            <label>MCDU web on browser/tablet/smartphone</label>
+        </text>
+
+        <button>
+            <halign>right</halign>
+            <pref-width>20</pref-width>
+            <pref-height>20</pref-height>
+            <legend>X</legend>
+            <key>Esc</key>
+            <binding>
+                <command>dialog-close</command>
+            </binding>
+        </button>
+    </group>
+	
+    <hrule/>
+	
+	<group>
+		<layout>vbox</layout>
+		
+		<text>
+			<halign>left</halign>
+			<label>Please check: HTTP service running is required.</label>
+			<visible>
+				<equals>
+					<property>/sim/http/running</property>
+					<value>0</value>
+				</equals>
+			</visible>			
+		</text>
+		<text>		
+			<halign>left</halign>
+			<label>Start FlightGear with -httpd=8080</label>
+			<visible>
+				<equals>
+					<property>/sim/http/running</property>
+					<value>0</value>
+				</equals>
+			</visible>			
+		</text>
+
+    </group>
+	
+	<group>
+		<layout>vbox</layout>
+
+		<text>
+			<halign>center</halign>
+			<label>Open MCDU on browser</label>
+		</text>
+
+		<button>
+			<halign>center</halign>
+			<legend>launch browser</legend>
+			<binding>
+				<command>nasal</command>
+				<script>
+					var n = getprop("/sim/http/running",0);
+					if( n != 1 ) {
+					  gui.popupTip("Internal webserver not running. Restart FlightGear with -httpd=8080", 5.0);
+					} else {
+					  var _url = "http://localhost:" ~ getprop("/sim/http/options/listening-port") ~ "/aircraft-dir/WebPanel/mcdu.html";
+					  fgcommand("open-browser", props.Node.new({ "url": _url }));
+					}				
+				</script>
+			</binding>
+		</button>
+	</group>	
+	
+	<group>
+		<layout>vbox</layout>
+		<text>
+			<halign>center</halign>
+			<label></label>
+		</text>
+		<text>
+			<halign>center</halign>
+			<label>Scan QR code with your smartphone or tablet</label>
+		</text>
+
+	</group>
+		
+	<canvas>
+		<name>mcduwebqr</name>
+		<valign>center</valign>
+		<halign>center</halign>
+		<stretch>false</stretch>
+		<pref-width>320</pref-width>
+		<pref-height>320</pref-height>
+		<nasal>
+			<load>
+				<![CDATA[
+				    var n = props.globals.getNode("/sim/http/running");
+					var mcdu_canvas_dlg = canvas.get(cmdarg());
+					var root = mcdu_canvas_dlg.createGroup();
+					mcdu_canvas_dlg.setColorBackground(1, 1, 1, 1);
+					var _url = "";
+					if( getprop("sim/http/running",0) == 1 ) _url = "http://localhost:" ~ getprop("/sim/http/options/listening-port") ~ "/aircraft-dir/WebPanel/mcdu.html";
+					root.createChild("image").set("src", "https://qr.eletto.dev/" ~ _url).setSize(300,300).setTranslation(10,10); 					
```

Comment:
I don't think, we should query such a service without the users consent (only generate that qr code if the user clicks a button like `Show QR code`

Reply:
Needed because port it's user defined.
Do you think it is a concern to use external service?
Maps and weathers work with external service too.

Reply:
Yes, cause it contains sensetiv information such as open ports and IP addresses...
Open ports are a potential security risk... so you shouldn't "announce" them to some service on the internet (except you know what you're doing -> not the case for normal users)

Reply:
I think most people will only need it on local network. If they need it outside, they should be able to figure it out themselves. I can't think of a use case for having it outside your LAN.

Reply:
True, but it doesn't mean, you should announce those things to the public... 
Remember stuff like the wannacry stuff.... it was only so successful cause people had their damn ports open.

Reply:
@merspieler not our situation. The port opens it's the same for phi.
As you request for a button, I added button to generate qrcode on last version,

Reply:
thanks